src/realm/sync/noinst/protocol_codec.hpp

```diff
@@ -491,6 +494,80 @@ class ClientProtocol {
         return action::ApplicationBug;
     }
 
+    template <typename Connection>
+    void parse_log_message(Connection& connection, HeaderLineParser& msg)
+    {
+        auto report_error = [&](const auto fmt, auto&&... args) {
+            auto msg = util::format(fmt, std::forward<decltype(args)>(args)...);
+            connection.handle_protocol_error(Status{ErrorCodes::SyncProtocolInvariantFailed, std::move(msg)});
+        };
+
+        auto session_ident = msg.read_next<session_ident_type>();
+        auto message_length = msg.read_next<size_t>('\n');
+        auto message_body_str = msg.read_sized_data<std::string_view>(message_length);
+        nlohmann::json message_body;
+        try {
+            message_body = nlohmann::json::parse(message_body_str);
+        }
+        catch (const nlohmann::json::exception& e) {
+            return report_error("Malformed json in log_message message: \"%1\": %2", message_body_str, e.what());
+        }
+        static const std::unordered_map<std::string_view, util::Logger::Level> name_to_level = {
+            {"fatal", util::Logger::Level::fatal},   {"error", util::Logger::Level::error},
```

Comment:
does the "fatal" log level terminate the process? seems risky to accept "fatal" logs from the server if so

Reply:
No - it is more related to the log level that specifies what is printed and does not affect the connection (or really anything else).
AFAIK, I don't think we are using the Fatal log level in any of our current debug log statements. The default log output level is Info, which will show Info, Error and Fatal log messages and all other levels are ignored.

Reply:
got it, just wanted to make sure because sometimes "fatal" is an alias for "error, then exit 1"